scripts/publish_binary_size.js

```diff
@@ -106,28 +111,62 @@ function query(after) {
 
                 row[i + 1] = run ? +run.summary.match(/is (\d+) bytes/)[1] : undefined;
             }
-
             rows.push(row);
         }
 
         if (history.pageInfo.hasNextPage) {
             return query(history.pageInfo.endCursor);
         } else {
-            return new AWS.S3({region: 'us-east-1'}).putObject({
-                Body: zlib.gzipSync(JSON.stringify(rows.reverse())),
-                Bucket: 'mapbox',
-                Key: 'mapbox-gl-native/metrics/binary-size/data.json',
-                ACL: 'public-read',
-                CacheControl: 'max-age=300',
-                ContentEncoding: 'gzip',
-                ContentType: 'application/json'
-            }).promise();
+          const latestRun = rows[0]
+          const runSizeMeasurements = latestRun.slice(1)
+
+          for (let i = 0; i < platforms.length; i++) {
+              const {platform, arch} = platforms[i];
+
+              sizeCheckInfo.push(JSON.stringify({
+                  'sdk': 'maps',
+                  'platform' : platform,
+                  'arch': arch,
+                  'size' : runSizeMeasurements[i],
+                  'created_at': formattedCurrentDate
+              }));
+          }
         }
     });
 }
 
 github.apps.createInstallationToken({installation_id: SIZE_CHECK_APP_INSTALLATION_ID})
     .then(({data}) => {
-        github.authenticate({type: 'token', token: data.token});
-        return query();
-    });
+      github.authenticate({type: 'token', token: data.token});
+
+      return query().then(function() {
+        // Uploads to data source used by 
+        // http://mapbox.github.io/mapbox-gl-native/metrics/binary-size/
+        var firstPromise = new AWS.S3({region: 'us-east-1'}).putObject({
+            Body: zlib.gzipSync(JSON.stringify(rows.reverse())),
+            Bucket: 'mapbox',
+            Key: 'mapbox-gl-native/metrics/binary-size/data.json',
+            ACL: 'public-read',
+            CacheControl: 'max-age=300',
+            ContentEncoding: 'gzip',
+            ContentType: 'application/json'
+        }).promise();
+        // Uploads to data source used by Mapbox internal metrics dashboards
+        var secondPromise = new AWS.S3({region: 'us-east-1'}).putObject({
+            Body: zlib.gzipSync(sizeCheckInfo.join('\n')),
+            Bucket: 'mapbox-loading-dock',
+            Key: `raw/nadia_staging_test_v6/${formattedCurrentDate}/mapbox-maps-ios-android-${process.env['CIRCLE_SHA1']}.json.gz`,
+            CacheControl: 'max-age=300',
+            ContentEncoding: 'gzip',
+            ContentType: 'application/json'
+        }).promise();
+        
+        return Promise.all([firstPromise, secondPromise]).then(data => {
+          console.log("Successfully uploaded all binary size metrics to S3");
```

Comment:
The success and error blocks in the `then` clause here do not return any values or chained Promise. That breaks the promise chain and ends up not returning the final result to the top-level.